src/pubsub/mod.rs

```diff
@@ -2,12 +2,86 @@ use crate::clients::*;
 use crate::workload::Component;
 use crate::workload::PublisherWorkItem as WorkItem;
 use crate::*;
+use ahash::RandomState;
 use async_channel::Receiver;
+use clocksource::Nanoseconds;
 use std::io::{Error, ErrorKind, Result};
 use tokio::runtime::Runtime;
 
+mod kafka;
 mod momento;
 
+struct MessageStamp {
+    hash_builder: RandomState,
+}
+pub enum MessageValidator {
```

Comment:
Maybe we can come up with a better name here: `MessageValidator` sounds like a struct, not an enum.

Reply:
Maybe MessageValidationResult?